### ISSUE TITLE
useResponsiveComponent prop values not applied during SSR

### ISSUE THREAD

**Issue by kettanaito** [kettanaito/atomic-layout#272]
## What:

```jsx
import styled from 'styled-components'
import { useResponsiveComponent } from 'atomic-layout'

const StyledImage = styled.img`
  border-radius: 5px;
`
const ElementImage = useResponsiveComponent(StyledImage)

const Example = () => (
  <ElementImage src="foo.png" />
)
```

The `ElementImage` component will render an image without the `src` attribute during SSR.

## Why:

My initial guess is that `useResponsiveComponent`, has its values throttled in a way which results the initial value not being set at all. 

## Environment:

- node vesrion: `12.x`
- npm version: `6.x`
- atomic-layout version: `0.12.x`

**Comment by kettanaito** [comment]
Would you have time to look into this issue please, @ruhaise? I'd gladly provide with assistance along the way.

**Comment by ruhaise** [comment]
@kettanaito im on it :)

**Comment by kettanaito** [comment]
@ruhaise Thanks! I can recommend to start from writing a unit test for SSR of `useResponsiveComponent` use case described above. Take a look at the [SSR unit test](https://github.com/kettanaito/atomic-layout/blob/bb37c3b28d99d8a6728e7e11f4e4ea9a820ba52f/packages/atomic-layout/test/ssr.spec.tsx) for inspiration. Perhaps we would have to extend that test suite to render responsive component for verification of this issue.

Seeing a relevant test failing is a great first step to start fixing anything.

**Comment by ruhaise** [comment]
@kettanaito thanks for the advice, sounds promising

**Comment by kettanaito** [comment]
Initial fix published in `atomic-layout@0.12.4`. Please update for it to take effect.

There is a new behavior-related issue found and will be addressed under #284.
